api/v1alpha1/thanosstore_types.go

```diff
@@ -0,0 +1,117 @@
+	// The idea of ignore-deletion-marks-delay is to ignore blocks that are marked for deletion with some delay.
+	// This ensures store can still serve blocks that are meant to be deleted but do not have a replacement yet.
+	// If delete-delay duration is provided to compactor or bucket verify component, it will upload deletion-mark.json
+	// file to mark after what duration the block should be deleted rather than deleting the block straight away.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="24h"
+	IgnoreDeletionMarksDelay Duration `json:"ignoreDeletionMarksDelay,omitempty"`
+	// YAML file that contains index cache configuration. See format details: https://thanos.io/tip/components/store.md/#index-cache
+	// IN-MEMORY config is loaded by default if not specified.
+	// +kubebuilder:validation:Optional
+	IndexCacheConfig corev1.ConfigMapKeySelector `json:"indexCacheConfig,omitempty"`
+	// YAML that contains configuration for caching bucket.
+	// See format details: https://thanos.io/tip/components/store.md/#caching-bucket"
+	// IN-MEMORY config is loaded by default if not specified.
+	// +kubebuilder:validation:Optional
+	CachingBucketConfig corev1.ConfigMapKeySelector `json:"cachingBucketConfig,omitempty"`
+	// ShardingStrategy defines the sharding strategy for the Store Gateways across object storage blocks.
+	// +kubebuilder:validation:Required
+	ShardingStrategy ShardingStrategy `json:"shardingStrategy,omitempty"`
+	// Minimum time range to serve. Any data earlier than this lower time range will be ignored.
+	// +kubebuilder:validation:Optional
+	MinTime Duration `json:"minTime,omitempty"`
+	// Maximum time range to serve. Any data after this upper time range will be ignored.
+	// +kubebuilder:validation:Optional
+	MaxTime Duration `json:"maxTime,omitempty"`
+}
+
+type ShardingStrategyType string
+
+const (
+	// Block is the block modulo sharding strategy for sharding Stores according to block ids.
+	Block ShardingStrategyType = "block"
+)
+
+// TODO(saswatamcode): Figure out sane default behaviour
+type ShardingStrategy struct {
+	// Type here is the type of sharding strategy.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default="block"
+	// +kubebuilder:validation:Enum=block
+	Type ShardingStrategyType `json:"type,omitempty"`
+	// Shards is the number of shards to split the data into.
+	// +kubebuilder:validation:Minimum=3
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Optional
+	Shards int32 `json:"shards,omitempty"`
+	// ReplicaPerShard is the number of replicas per shard.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Optional
+	ReplicaPerShard int32 `json:"replicaPerShard,omitempty"`
+	// BlockModulo is the modulo value to use for block based sharding strategy.
+	// +kubebuilder:validation:Minimum=6
+	// +kubebuilder:validation:Optional
+	BlockModulo int32 `json:"blockModulo,omitempty"`
```

Comment:
wouldnt this be inferred from the "shard count" above?

Reply:
Yes, I just added min, also if user wants to override for some reason. Will remove the min